### DISCUSSION TITLE
Better health hazards warnings

### DISCUSSION THREAD

**Discussion by Slayer95** [ppy/osu#13027] [Ideas]
**Describe the new feature:** 
Improve coverage, specificity and visibility of gameplay health hazards.

**Proposal designs of the feature:** 

# Coverage
Specific osu! maps currently support opt-in epileptic seizure warnings. However, other conditions that might be triggered or created during osu gameplay include (but may not be limited to):
- Vestibular disorders ("motion sickness"). See e.g. https://www.w3.org/WAI/WCAG21/Understanding/animation-from-interactions.html and references therein.
- Carpal tunnel syndrome (CTS)

# Specificity
Currently, epileptic seizure warnings are displayed in a per-map basis. This is great, if compared to an approach that would show a single warning when the game is opened.

On the other hand, the text of these warnings refers to the *video* as the source of trouble and asks the player to take caution. However, the source of trouble may actually be the *Storyboard*. Therefore, a battle-hardened player affected by epilepsy would immediately turn off *both* the video and the storyboard, but an unaware player may only turn off the video, with possible tragic consequences. Note also that some maps, such as the infamous [Centipede - Knife Party](https://osu.ppy.sh/beatmapsets/150945#osu/372245) might be triggers themselves even without video and storyboards (and yes, despite it says "it's not a map, just a visualization", people do download it and play and/or view it).

*Proposal*: Add support for mappers to flag specifically the video, storyboard, map, or any combination of them as triggers for epilepsy, as well as other illnesses mentioned in the **Coverage** section.

# Visibility
Health hazard warnings should be visible in the multiplayer room, song select menu, and in the playlist of a timeshift multiplayer room, so that the player has a large margin of time to react to it. In osu!stable, the player might not have enough time to turn off video AND storyboard before the game actually starts.

Furthermore, in the osu website, in the download screen, it should say something like "intensive gaming sessions of some osu! game modes may cause injury" (this is the least dramatical text for an initial warning re: CTS).

**Comment by peppy** [comment]
I'm pretty sure the ones you mention apply to every beatmap. Also if you suffer from epilepsy you would already be very careful with which games you play.

Flagging per item in the beatmap sounds unnecessarily complicated, too.

**Comment by Slayer95** [comment]
> Flagging per item in the beatmap sounds unnecessarily complicated, too.

I assume you mean storyboard/video/composition by "item" here. I think there would be value in making this distinction. However, a fair compromise would be fixing the text of the warnings so that they  don't imply that the corrective measure would be disabling video, but both video *and* storyboard.

> Also if you suffer from epilepsy you would already be very careful with which games you play.

That's true, but I don't see how that affects this proposal.

> I'm pretty sure the ones you mention apply to every beatmap.

- CTS: I think the claim that playing std maps in Easy or Normal difficulties for short-to-medium-length gaming sessions may trigger carpal tunnel syndrome would require evidence.

- Motion sickness: While it's true that every osu! game mode inherently has elements that might trigger motion sickness, these elements are different for each game mode, and their impact vary in a per-user-basis, and may be manipulated by the user.

In particular, I should note that approach circles in std may be removed altogether by turning HD on. Regarding approach circles: I actually suffer from motion sickness myself; however, I almost never get disoriented by them. On the other hand, some storyboards or videos introduce further triggers that might be unexpected *a priori*. **What I would expect from this proposal is for these "further triggers" to be flagged.** For example, the following maps:

A. [CROSSxBEATS (Ocelot) - YUKARI [Mikiwak's Insane]](https://osu.ppy.sh/beatmapsets/497536#osu/1429835)
- *Element*: Storyboard
- *Gameplay*: https://youtu.be/71Vd622GAEs
- (This map happens to warn for epilepsy)

In https://youtu.be/71Vd622GAEs?t=27, the storyboard shows a pink flower with wisps of light rotating around it both clockwise and counterclockwise. *That's disorienting*. Furthermore, in https://youtu.be/71Vd622GAEs?t=50, sakura petals start flying in from the left side of the screen towards the right side. *That's a third background movement in its own direction*

These effects fullfill the three big trigger criteria listed in [esigning Safer Web Animation For Motion Sensitivity](http://alistapart.com/article/designing-safer-web-animation-for-motion-sensitivity/#section4):
1. (Big) Relative size of movement.
2. Mismatched directions and speed.
3. (Large) Distance covered.

B. [fhana - Rhapsody of Blue Sky - [Nozely's Easy]](https://osu.ppy.sh/beatmapsets/556757#osu/1200812)
- *Element*: Video
- *Gameplay*: https://youtu.be/uFOJmFkKb2k
- (This map doesn't have any warning)

Uhm this video has a lot of potential motion sickness triggers... Just to list a few:
- Wiggling? choreography in https://youtu.be/uFOJmFkKb2k?t=31
- Duplication of character and bidirectional movement in https://youtu.be/uFOJmFkKb2k?t=53
- Accelerated rotation of many objects all over the screen in https://youtu.be/uFOJmFkKb2k?t=55
- Bidirectional movement again in https://youtu.be/uFOJmFkKb2k?t=70
- Not enough trouble in the previous item? Make the girl and many clones of her run towards you in https://youtu.be/uFOJmFkKb2k?t=73

**Comment by peppy** [comment]
Question: do you see every other game on the market have warnings about every possible thing they could cause? Maybe in disclaimers hidden in some instruction manual for large companies, but in general you'll find the answer is no.

**Comment by Slayer95** [comment]
What makes osu! special is that it's a game reliant on user-submitted content. In that context, I'd argue there is a responsibility from the game staff to implement policies geared to analyzing this content in order to protect users. Quoting the website I linked above:

> approximately 8 million American adults report a chronic problem with balance, while an additional 2.4 million report a chronic problem with dizziness.

I'd think that since osu! already warns for epilepsy, adding just another per-map warning category wouldn't be too much to ask for? Of course, that would make map moderation harder, but ranking criteria is already strict. And again, *responsibility*.

Carpal tunnel syndrome is a different beast that would need a separate treatment, since it's not per-map, but kinda all-over the place. Still, I am certain that its current handling is very unsatisfactory. I mean, people *are* getting hurt playing osu!, and I would dare to claim that some of them might not have even known what CTS was before the harm is done.

Of course, I think you have (for now) every legal right to dismiss this concern and close as wontfix...

**Comment by peppy** [comment]
People that have these issues know they have the issues. We don't need to warn them.

As for stress related injuries, we do plan on adding warnings every so often which tell the user to take a break (and there's also https://github.com/ppy/osu/issues/3352 to cover this to some extent).

**Comment by Slayer95** [comment]
> People that have these issues know they have the issues. We don't need to warn them.

What? I don't need you to tell me that motion sickness exists. I can play osu! easily for hours or days in multiplayer without issues, but a map with lots of motion sickness triggers is bound to be selected at some point, and I would be very grateful if you'd let me know whether it has these in advance, so that I can turn off elements, or simply skip it.

> As for stress related injuries, we do plan on adding warnings every so often which tell the user to take a break (and there's also #3352 to cover this to some extent).

This is great, thanks for considering it!
